# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/AudioTrack.h
# Commit: c54cb3c10992
# Full Hash: c54cb3c109922bc15fb6d0ca67e5f2a05980c1a5
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-07-31 15:46:43
# Regressor Bug: 1493613
# File Overlap Count: 2
# Description:
#   Bug 1493613 - Move MediaStream control from DOMMediaStream to MediaStreamTrack. r=padenot
#   
#   This is inherently large, because modifying these bits of DOMMediaStream and
#   MediaStreamTrack affects all consumers and producers of all DOMMediaStreams and
#   MediaStreamTracks.
# ==============================================================================

diff -r 0b03dd9d20ac -r c54cb3c10992 dom/media/AudioTrack.h
--- a/dom/media/AudioTrack.h	Wed Jul 31 09:35:06 2019 +0000
+++ b/dom/media/AudioTrack.h	Wed Jul 31 07:58:17 2019 +0000
@@ -12,11 +12,17 @@
 namespace mozilla {
 namespace dom {
 
+class AudioStreamTrack;
+
 class AudioTrack : public MediaTrack {
  public:
   AudioTrack(nsIGlobalObject* aOwnerGlobal, const nsAString& aId,
              const nsAString& aKind, const nsAString& aLabel,
-             const nsAString& aLanguage, bool aEnabled);
+             const nsAString& aLanguage, bool aEnabled,
+             AudioStreamTrack* aStreamTrack = nullptr);
+
+  NS_DECL_ISUPPORTS_INHERITED
+  NS_DECL_CYCLE_COLLECTION_CLASS_INHERITED(AudioTrack, MediaTrack)
 
   JSObject* WrapObject(JSContext* aCx,
                        JS::Handle<JSObject*> aGivenProto) override;
@@ -25,13 +31,21 @@
 
   void SetEnabledInternal(bool aEnabled, int aFlags) override;
 
+  // Get associated audio stream track when the audio track comes from
+  // MediaStream. This might be nullptr when the src of owning HTMLMediaElement
+  // is not MediaStream.
+  AudioStreamTrack* GetAudioStreamTrack() { return mAudioStreamTrack; }
+
   // WebIDL
   bool Enabled() const { return mEnabled; }
 
   void SetEnabled(bool aEnabled);
 
  private:
+  virtual ~AudioTrack();
+
   bool mEnabled;
+  RefPtr<AudioStreamTrack> mAudioStreamTrack;
 };
 
 }  // namespace dom